Ftp/New-FtpDirectory/New-FtpDirectory.psm1

```diff
@@ -0,0 +1,154 @@
+            $destinationFolder = $folder.Fullname -replace $srcFolderPath, $Url
+            $destinationFolder = $destinationFolder -replace "\\", "/"
+         
+            try
+            {
+                $makeDirectory = [System.Net.WebRequest]::Create($destinationFolder)
+                $makeDirectory.Credentials = $credentials
+                $makeDirectory.Method = [System.Net.WebRequestMethods+FTP]::MakeDirectory
+                $makeDirectory.GetResponse()
+    
+                Write-Host "Folder created successfully."
+                Write-Host "Destination folder:" $destinationFolder
+            }
+            catch [Net.WebException]
+            {
+                try
+                {
+                    $checkDirectory = [System.Net.WebRequest]::Create($destinationFolder)
+                    $checkDirectory.Credentials = $credentials
+                    $checkDirectory.Method = [System.Net.WebRequestMethods+FTP]::PrintWorkingDirectory
+                    $checkDirectory.GetResponse()
+                    
+                    Write-Host "Folder already exists."
+                    Write-Host "Destination folder:" $destinationFolder
+                }
+                catch [Net.WebException]
+                {
+                    throw "Other error encountered during subfolders creation."
+                }
+            }
+        }
+         
+        # Upload files.
+        $webclient = New-Object System.Net.WebClient
+        $webclient.Credentials = $credentials
+    
+        foreach ($file in $srcFiles)
+        {
+            $srcFullPath = $file.fullname
+            $srcFilePath = $LocalFolderPath -replace "\\", "\\" -replace "\:", "\:"
+            $destinationFile = $srcFullPath -replace $srcFilePath, $Url
+            $destinationFile = $destinationFile -replace "\\", "/"
+         
+            $uri = New-Object System.Uri($destinationFile) 
+           
+            Write-Host "Uploading file:" $srcFullPath
+            Write-Host "File uri:" $uri
+            $errorCount = 0
+    
+            do
+            {
+                try
+                {
+                    $webclient.UploadFile($uri, $srcFullPath)
+                    Write-Host "Upload successful."
+    
+                    break
+                }
+                catch
+                {
+                    Write-Host "Error caught, trying initializing new webclient object."
+                    $errorCount++
+                    Write-Host "ERROR COUNT:" $errorCount
+    
+                    $webclient.Dispose()
+                    Start-Sleep -s 5
+    
+                    $webclient = New-Object System.Net.WebClient 
+                    $webclient.Credentials = $credentials 
+                }
+            } while ($errorCount -ne 10)
+    
+            if ($errorCount -eq 10) 
+            {
+                throw "Maximum error count exceeded."
+            }
+        }
+    
+        $webclient.Dispose()
```

Comment:
Dispose()-t mindig `Finally`-ban kellene, a többi helyen is (ha már nincs `using` PS-ben).

Reply:
`try { } finally { }`-be rakom ezeket. Pl:

![image](https://user-images.githubusercontent.com/51411356/123108572-d0ba6400-d43a-11eb-823d-81d315507017.png)

Reply:
Cool!